src/commands/inputs.rs

```diff
@@ -0,0 +1,279 @@
+        }
+
+        let v: &wdl::analysis::types::Type = input.ty();
+        let key = format!("{}.{}", parent_name, name);
+        let value = type_to_json(&v);
+        template.insert(key, value);
+    }
+
+    let json_output = serde_json::to_string_pretty(&template)?;
+
+    if let Some(output_path) = args.output {
+        std::fs::write(output_path, json_output)?;
+    } else {
+        println!("{}", json_output);
+    }
+
+    Ok(())
+}
+
+fn type_to_json(ty: &Type) -> Value {
+    match ty {
+        Type::Primitive(ty, _bool) => match ty {
+            wdl::analysis::types::PrimitiveType::Boolean => Value::String("Boolean".to_string()),
+            wdl::analysis::types::PrimitiveType::Integer => Value::String("Integer".to_string()),
+            wdl::analysis::types::PrimitiveType::Float => Value::String("Float".to_string()),
+            wdl::analysis::types::PrimitiveType::String => Value::String("String".to_string()),
+            wdl::analysis::types::PrimitiveType::File => Value::String("File".to_string()),
+            wdl::analysis::types::PrimitiveType::Directory => {
+                Value::String("Directory".to_string())
+            }
+        },
+
+        _ => Value::Null,
+    }
+}
+
+fn has_default(document: &Document, document_path: &str, input_name: &str) -> bool {
+    let ast_doc: AstDocument = document.node();
+
+    let mut visitor = InputVisitor {
+        inputs: IndexMap::new(),
+        literal_defaults: IndexMap::new(),
+    };
+
+    ast_doc.visit(&mut (), &mut visitor);
+
+    visitor.inputs.get(input_name).cloned().unwrap_or(false)
+}
+
+fn has_literal_default(document: &Document, document_path: &str, input_name: &str) -> bool {
+    let ast_doc: AstDocument = document.node();
+
+    let mut visitor = InputVisitor {
+        inputs: IndexMap::new(),
+        literal_defaults: IndexMap::new(),
+    };
+
+    ast_doc.visit(&mut (), &mut visitor);
+
+    visitor
+        .literal_defaults
+        .get(input_name)
+        .cloned()
+        .unwrap_or(false)
+}
+
+// Collects inputs with their parent names
+fn collect_inputs_with_parents<'a>(
+    args: &'a InputsArgs,
+    document: &'a Document,
+) -> Result<Vec<(&'a str, &'a str, &'a Input)>> {
+    let mut result: Vec<(&str, &str, &Input)> = Vec::new();
+
+    if let Some(name) = &args.name {
+        // Specific task or workflow requested
+        if let Some(task) = document.task_by_name(name) {
+            for (input_name, input) in task.inputs() {
+                result.push((task.name(), input_name, input));
+            }
+        } else if let Some(workflow) = document.workflow() {
+            if workflow.name() == name {
+                // Add workflow inputs
+                for (input_name, input) in workflow.inputs() {
+                    result.push((workflow.name(), input_name.as_str(), input));
+                }
+
+                // If nested_inputs is true, add all called task inputs
+                if args.nested_inputs {
```

Comment:
This is not the intended behavior for `--nested-inputs`. "nested inputs" refer to this section of the WDL spec - https://github.com/openwdl/wdl/blob/wdl-1.2/SPEC.md#computing-call-inputs

> Any optional inputs that are not specified in the call may be set by the user at runtime if the execution engine supports the allow_nested_inputs hint and it is set to true in the workflow's hints section.

To support nested inputs in this command, you will need to iterate through a workflows "calls" and extract the relevant inputs from the called tasks and subworkflows.